src/aiovodafone/api.py

```diff
@@ -145,33 +149,45 @@ async def _post_page_result(
         page: str,
         payload: dict[str, Any],
         timeout: int = 10,
-    ) -> ClientResponse:
+    ) -> ClientResponse | None:
         """Get data from a web page via POST."""
-        _LOGGER.debug("POST page  %s from host %s", page, self.host)
+        _LOGGER.debug("POST page %s from host %s", page, self.host)
         timestamp = int(datetime.now(tz=UTC).timestamp())
         url = f"{self.base_url}{page}?_={timestamp}&csrf_token={self.csrf_token}"
-        return await self.session.post(
-            url,
-            data=payload,
-            headers=self.headers,
-            timeout=timeout,
-            ssl=False,
-            allow_redirects=True,
-        )
+        try:
+            return await self.session.post(
+                url,
+                data=payload,
+                headers=self.headers,
+                timeout=ClientTimeout(timeout),
+                ssl=False,
+                allow_redirects=True,
+            )
+        except ClientResponseError:
+            _LOGGER.exception("Request to %s failed", page)
```

Comment:
I would suggest to log the specific exception as well

```suggestion
        except ClientResponseError as err:
            _LOGGER.exception("Request to %s failed: %s", page, repr(err))
```